pkg/kubelet/kubelet.go

```diff
@@ -2840,17 +2840,13 @@ func isPodResizeInProgress(pod *v1.Pod, podStatus *kubecontainer.PodStatus) bool
 // Returns true if the resize can proceed.
 func (kl *Kubelet) canResizePod(pod *v1.Pod) (bool, v1.PodResizeStatus) {
 	if v1qos.GetPodQOS(pod) == v1.PodQOSGuaranteed && !utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScalingExclusiveCPUs) {
-		if utilfeature.DefaultFeatureGate.Enabled(features.CPUManager) {
-			if kl.containerManager.GetNodeConfig().CPUManagerPolicy == "static" {
-				klog.V(3).InfoS("Resize is infeasible for Guaranteed Pods alongside CPU Manager static policy")
-				return false, v1.PodResizeStatusInfeasible
-			}
+		if kl.containerManager.CanAllocateExclusively(v1.ResourceCPU) {
+			klog.V(3).InfoS("Resize is infeasible for Guaranteed Pods alongside CPU Manager static policy")
+			return false, v1.PodResizeStatusInfeasible
 		}
-		if utilfeature.DefaultFeatureGate.Enabled(features.MemoryManager) {
-			if kl.containerManager.GetNodeConfig().ExperimentalMemoryManagerPolicy == "static" {
```

Comment:
cc @tallclair due to historycal reasons, memory manager policy names are CamelCase (e.g. "Static") so you may have a hidden bug here. My PR wants to remove this class of possible bugs.